backend/danswer/connectors/factory.py

```diff
@@ -24,45 +25,52 @@ def build_connector(
     source: DocumentSource,
     input_type: InputType,
     connector_specific_config: dict[str, Any],
-) -> PullLoader | RangePullLoader:
+) -> BaseConnector:
     if source == DocumentSource.SLACK:
-        if input_type == InputType.PULL:
-            return PeriodicSlackLoader(**connector_specific_config)
-        if input_type == InputType.LOAD_STATE:
-            return BatchSlackLoader(**connector_specific_config)
+        connector: BaseConnector = SlackConnector(**connector_specific_config)
     elif source == DocumentSource.GOOGLE_DRIVE:
-        if input_type == InputType.PULL:
-            return BatchGoogleDriveLoader(**connector_specific_config)
+        connector = GoogleDriveConnector(**connector_specific_config)
     elif source == DocumentSource.GITHUB:
-        if input_type == InputType.PULL:
-            return BatchGithubLoader(**connector_specific_config)
+        connector = GithubConnector(**connector_specific_config)
     elif source == DocumentSource.WEB:
-        if input_type == InputType.PULL:
-            return WebLoader(**connector_specific_config)
+        connector = WebConnector(**connector_specific_config)
+    else:
+        raise ConnectorMissingException(f"Connector not found for source={source}")
 
-    raise ConnectorMissingException(
-        f"Connector not found for source={source}, input_type={input_type}"
-    )
+    if any(
+        [
+            input_type == InputType.LOAD_STATE
+            and not isinstance(connector, LoadConnector),
+            input_type == InputType.POLL and not isinstance(connector, PollConnector),
+            input_type == InputType.EVENT and not isinstance(connector, EventConnector),
+        ]
+    ):
+        raise ConnectorMissingException(
+            f"Connector for source={source} does not accept input_type={input_type}"
+        )
 
-
-def build_pull_connector(
-    source: DocumentSource, connector_specific_config: dict[str, Any]
-) -> PullLoader:
-    connector = build_connector(source, InputType.PULL, connector_specific_config)
-    return (
-        _range_pull_to_pull(connector)
-        if isinstance(connector, RangePullLoader)
-        else connector
-    )
+    return connector
 
 
-def _range_pull_to_pull(range_pull_connector: RangePullLoader) -> PullLoader:
-    class _Connector(PullLoader):
+# TODO this is some jank, rework at some point
+def _pull_to_load_connector(range_pull_connector: PollConnector) -> LoadConnector:
```

Comment:
`_pull_to_load_connector` should be `_poll_to_load_connector`?

Reply:
done